doc/shared/sunadjoint/SUNAdjointStepper.rst

```diff
@@ -0,0 +1,153 @@
+..
+   ----------------------------------------------------------------
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2025, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNAdjoint.Stepper:
+
+The SUNAdjointStepper Class
+===========================
+
+.. versionadded:: x.y.z
+
+.. c:type:: SUNAdjointStepper
+
+   The :c:type:`SUNAdjointStepper` class provides a package-agnostic interface to
+   SUNDIALS ASA capabilities. It currently only supports the discrete ASA
+   capabilities in the ARKODE package, but in the future this support may be
+   expanded.
+
+Class Methods
+-------------
+
+The :c:type:`SUNAdjointStepper` class has the following methods:
+
+.. c:function:: SUNErrCode SUNAdjointStepper_Create(SUNStepper fwd_sunstepper, sunbooleantype own_fwd, \
+   SUNStepper adj_sunstepper, sunbooleantype own_adj, suncountertype final_step_idx, sunrealtype tf, N_Vector sf, \
+   SUNAdjointCheckpointScheme checkpoint_scheme, SUNContext sunctx, SUNAdjointStepper* adj_stepper)
+
+   Creates the ``SUNAdjointStepper`` object needed to solve the adjoint problem.
+
+   :param fwd_sunstepper: The :c:type:`SUNStepper` to be used for forward computations of the original ODE.
+   :param own_fwd: Should `fwd_sunstepper` be owned (and destroyed) by the `SUNAdjointStepper` or not.
+   :param adj_sunstepper: The :c:type:`SUNStepper` to be used for the backward integration of the adjoint ODE.
+   :param own_adj: Should `adj_sunstepper` be owned (and destroyed) by the `SUNAdjointStepper` or not.
+   :param final_step_idx: The index (step number) of the step corresponding to ``t_f`` for the forward ODE.
+   :param tf: The terminal time for the forward ODE (the initial time for the adjoint ODE).
+   :param sf: The terminal condition for the adjoint ODE.
+   :param checkpoint_scheme: The :c:type:`SUNAdjointCheckpointScheme` object that determines the checkpointing strategy to use. This should be the same object provided to the forward integrator/stepper.
+   :param sunctx: The :c:type:`SUNContext` for the simulation.
+   :param adj_stepper: The :c:type:`SUNAdjointStepper` to construct (will be ``NULL`` on failure).
+
+   :return: A :c:type:`SUNErrCode` indicating failure or success.
+
+
+.. c:function:: SUNErrCode SUNAdjointStepper_ReInit(SUNAdjointStepper adj, N_Vector sf, sunrealtype tf)
```

Comment:
Missing `t0` and `y0` inputs. Also see comment about update the order to align with `t` first style: `SUNAdjointStepper_ReInit(adj, t0, y0, tf, sf)`

Reply:
This was updated on the ARKODE branch. Ill see if I can pull it into there.